imap_processing/codice/codice_l1a.py

```diff
@@ -408,14 +425,9 @@ def create_hskp_dataset(
     cdf_attrs.add_instrument_variable_attrs("codice", "l1a")
     cdf_attrs.add_global_attribute("Data_version", data_version)
 
-    metadata_arrays: dict = collections.defaultdict(list)
-
-    for packet in packets:
-        add_metadata_to_array(packet, metadata_arrays)
-
     epoch = xr.DataArray(
         met_to_j2000ns(
-            metadata_arrays["SHCOARSE"],
+            packet.shcoarse.data,
```

Comment:
This should already be taken care of in the "epoch" variable. (I wonder if we will run into issues with you here and the different reference_epoch...

Reply:
Thanks, good catch. So if I am understanding correctly, this can be changed from:

```python
    epoch = xr.DataArray(
        met_to_j2000ns(
            packet.shcoarse.data,
            reference_epoch=np.datetime64("2010-01-01T00:01:06.184", "ns"),
        ),
        name="epoch",
        dims=["epoch"],
        attrs=cdf_attrs.get_variable_attributes("epoch"),
    )
```

to:

```python
    epoch = xr.DataArray(
        packet.epoch,
        name="epoch",
        dims=["epoch"],
        attrs=cdf_attrs.get_variable_attributes("epoch"),
    )
```